backends/tofino/bf-p4c/arch/psa/rewrite_packet_path.cpp

```diff
@@ -437,9 +437,8 @@ struct PacketPath : public Transform {
         auto select = new IR::PathExpression(IR::ID("__egress_p4_entry_point"));
         auto newStateName = IR::ID(cstring("__mirror_" + gress_str));
         auto *newState = new IR::ParserState(newStateName, *statements, select);
-        newState->annotations = newState->annotations->addAnnotationIfNew(
-            IR::Annotation::nameAnnotation,
-            new IR::StringLiteral(cstring("$__mirror_" + gress_str)));
+        newState->annotations = IR::Annotations::maybeAddNameAnnotation(newState->annotations,
```

Comment:
Do you even need the assignment here? Couldn't we use `newState->addAnnotationIfNew`?

Reply:
Yes, though it might be a bit more verbose as we'd need to construct `IR::StringLiteral`